lib/GHCup/Download.hs

```diff
@@ -161,22 +155,17 @@ getDownloadsF :: ( FromJSONKey Tool
                    GHCupInfo
 getDownloadsF pfreq@(PlatformRequest arch plat _) = do
   Settings { urlSource } <- lift getSettings
-
-  -- Check for custom URL sources that aren't in the default list
   forM_ urlSource $ \src -> 
     case src of
       NewURI uri -> do
-        let url = show uri
-        when (not (url `elem` defaultURLs) && 
-              not (any (`isPrefixOf` url) officialPrefixes) ||
-              url == "https://ghc.gitlab.haskell.org/ghcup-metadata/ghcup-nightlies-0.0.7.yaml") $ 
-          logWarn $ "The URL " <> T.pack url <> " is not an official GHCup metadata source and may not be maintained or QA'd by GHCup."
+        when (not (isOfficialURI uri) || isNightliesURI uri) $ 
```

Comment:
1. we do not want to emit a warning every time a download is issued... instead we want to emit a warning if a user adds a new channel (either via `ghcup config add-release-channel` or `ghcup config set url-source`)
2. nightlies are no exception... they are not maintained by GHCup, so we want to warn